python/cuml/test/test_pickle.py

```diff
@@ -83,16 +91,63 @@ def pickle_save_load(tmpdir, model):
     return cu_after_pickle_model
 
 
+def make_classification_dataset(datatype, nrows, ncols, n_info):
+    X, y = make_classification(n_samples=nrows, n_features=ncols,
+                               n_informative=n_info,
+                               random_state=0, n_classes=2)
+    X = X.astype(datatype)
+    y = y.astype(np.int32)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.8)
+    return X_train, y_train, X_test
+
+
 def make_dataset(datatype, nrows, ncols, n_info):
     X, y = make_regression(n_samples=nrows, n_features=ncols,
                            n_informative=n_info, random_state=0)
     X = X.astype(datatype)
     y = y.astype(datatype)
     X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.8)
-
     return X_train, y_train, X_test
 
 
+@pytest.mark.parametrize('datatype', [np.float32])
+@pytest.mark.parametrize('model', rf_regression_model.values())
+@pytest.mark.parametrize('nrows', [unit_param(500)])
+@pytest.mark.parametrize('ncols', [unit_param(16)])
+@pytest.mark.parametrize('n_info', [unit_param(7)])
+def test_rf_regression_pickle(tmpdir, datatype, model, nrows, ncols, n_info):
+    X_train, y_train, X_test = make_dataset(datatype, nrows, ncols, n_info)
+
+    model.fit(X_train, y_train)
+    cu_before_pickle_predict = np.asarray(model.predict(X_test))
+
+    cu_after_pickle_model = pickle_save_load(tmpdir, model)
+
+    cu_after_pickle_predict = np.asarray(cu_after_pickle_model.predict(X_test))
+
+    assert array_equal(cu_before_pickle_predict, cu_after_pickle_predict)
+
+
+@pytest.mark.parametrize('datatype', [np.float32])
+@pytest.mark.parametrize('model', rf_classification_model.values())
+@pytest.mark.parametrize('nrows', [unit_param(20)])
+@pytest.mark.parametrize('ncols', [unit_param(7)])
+@pytest.mark.parametrize('n_info', [unit_param(5)])
+def test_rf_classification_pickle(tmpdir, datatype, model,
+                                  nrows, ncols, n_info):
+    X_train, y_train, X_test = make_classification_dataset(datatype, nrows,
+                                                           ncols, n_info)
+
+    model.fit(X_train, y_train)
+    cu_before_pickle_predict = np.asarray(model.predict(X_test))
+
+    cu_after_pickle_model = pickle_save_load(tmpdir, model)
+
+    cu_after_pickle_predict = np.asarray(cu_after_pickle_model.predict(X_test))
+
+    assert array_equal(cu_before_pickle_predict, cu_after_pickle_predict)
```

Comment:
Need to address issue #1309 which might be particularly relevant for RF since it might be susceptible to storing pointers accidentally style of bugs